### DISCUSSION TITLE
Few more lexers I am looking for

### DISCUSSION THREAD

**Discussion by mcepl** [orbitalquark/scintillua#98] [Q&A]
I am working on rebase of [vis](https://git.sr.ht/~martanne/vis) lexers to the [scintillua_6.2](https://github.com/orbitalquark/scintillua/releases/tag/scintillua_6.2) (current work is in my [devel_scintillua](https://git.sr.ht/~mcepl/vis/log/devel_scintillua) branch).

For purposes of testing I try to use vis everywhere and I have now encountered couple of situations where a lexer would be useful, but I cannot find any. I use currently blank template files, but it would be lovely to have something real.

1. vis (or any other scintillua-based editor) as editor for writing emails (in [mutt](http://www.mutt.org/) or [aerc](https://git.sr.ht/~rjarry/aerc)). It would be probably the equivalent of [mail.vim](https://github.com/vim/vim/blob/master/runtime/syntax/mail.vim).
2. vis as `core.editor` for git. Vim has:

- [git.vim](https://github.com/vim/vim/blob/master/runtime/syntax/git.vim): I am not sure what it is good for,
- [gitattributes.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitattributes.vim): tiny, not important,
- [gitcommit.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitcommit.vim): that would be lovely to have,
- [gitconfig.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitconfig.vim): not sure it is really needed, a generic [ini.lua](https://github.com/orbitalquark/scintillua/blob/default/lexers/ini.lua) seems to work just fine,
- [gitignore.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitignore.vim): tiny, unimportant,
- [gitrebase.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitrebase.vim): we have [git-rebase.lua](https://github.com/orbitalquark/scintillua/blob/default/lexers/git-rebase.lua), and it works just fine, and
- [gitsendemail.vim](https://github.com/vim/vim/blob/master/runtime/syntax/gitsendemail.vim): that seems to be just the tinyiest layer over the general mail.vim I mentioned above.

Would anybody have at least some drafts of these files, please?

Thank you,

Matěj
-- 
https://matej.ceplovi.cz/blog/, @mcepl@floss.social
GPG Finger: 3C76 A027 CA45 AD70 98B5  BC1D 7920 5802 880B C9D8
 
Blessed be the God […] who comforts us in all our affliction, so
that we may be able to comfort those who are in any affliction,
with the comfort with which we ourselves are comforted by God.
  -- 2. Corinthians 1:3-4

**Comment by samuelmarquis** [comment]
All of these would be useful for me as well, I wouldn't mind taking a stab at some of them if you haven't yet.
